### PR TITLE
Subscription loaders

### PR DESCRIPTION
**Pull request by kidqueb**
This PR actually reverts the previous one a bit. The issue before was that if we defined loaders, they weren't passed into subscriptions. This made it so even if we passed through all the required data it would throw an error and overwrite the data given with null.

Now that we have loaders being passed in that isn't a possibility. I tried to force the issue in some tests and do some cheeky stuff like passing `undefined` as a loader but it would error long before getting through loader definition.

### CODE REVIEW

**Review by jonnydgreen — COMMENTED** [review]
Looks awesome!

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
You should this be landed in place of https://github.com/mercurius-js/mercurius/pull/512?

File: index.js [hunk 1]
```diff
@@ -358,12 +359,18 @@ const plugin = fp(async function (app, opts) {
       app.decorate(kFactory, factory)
     }
 
-    function defineLoader (name) {
+    if (!subscriptionFactory) {
+      subscriptionFactory = new Factory()
+      app.decorate(kSubscriptionFactory, subscriptionFactory)
+    }
+
+    function defineLoader (name, prop) {
```
[the remark this answers is not in the document]
Comment (kidqueb) [reply index.js]: Good catch. Killed it 😬

File: index.js [hunk 2]
```diff
@@ -374,11 +381,13 @@ const plugin = fp(async function (app, opts) {
       resolvers[typeKey] = {}
       for (const prop of Object.keys(type)) {
         const name = typeKey + '-' + prop
-        resolvers[typeKey][prop] = defineLoader(name)
+        resolvers[typeKey][prop] = defineLoader(name, prop)
```
[the remark this answers is not in the document]
Comment (kidqueb) [reply index.js]: > You should this be landed in place of #512?

Yessir 👍

**Comment by kidqueb** [comment]
> You should this be landed in place of #512?

Yessir 👍

**Review by jonnydgreen — APPROVED** [review]
LGTM!

**Pull request merged**